File: arena-junit/src/main/java/arena/junit/playbook/ActivePostgresPlaybook.java
package arena.junit.playbook;

import arena.junit.ffi.ArenaBindings;
import arena.junit.ffi.ArenaBindingError;
import arena.junit.support.ArenaJson;

import com.fasterxml.jackson.databind.node.ObjectNode;
import com.sun.jna.Pointer;

public class ActivePostgresPlaybook extends ActivePlaybook {
  protected ActivePostgresPlaybook(Pointer handle) {
    super(handle);
  }

  public void verify(String query, int expectedValue) {
    ObjectNode spec = ArenaJson.object();
    spec.put("query", query);
    spec.put("expected_value", expectedValue);
    try {
      ArenaBindings.postgresPlaybookVerify(handle(), ArenaJson.MAPPER.writeValueAsString(spec));
    } catch (ArenaBindingError e) {
      noteBodyFailure();
      throw e;
    } catch (Exception e) {
      throw new ArenaBindingError(e.getMessage());
    }
  }
}

File: arena-xunit/src/Playbook/ActivePostgresPlaybook.cs
using System;
using ArenaDotnet.Xunit.Ffi;
using ArenaDotnet.Xunit.Support;
using Newtonsoft.Json;

namespace ArenaDotnet.Xunit.Playbook;

public sealed class ActivePostgresPlaybook : ActivePlaybook
{
    public ActivePostgresPlaybook(IntPtr handle) : base(handle) { }

    public void Verify(string query, int expectedValue)
    {
        var spec = ArenaJson.Serialize(new PostgresVerifySpec
        {
            Query = query,
            ExpectedValue = expectedValue,
        });
        ArenaBindings.PostgresPlaybookVerify(_handle, spec);
    }

    [JsonObject(ItemNullValueHandling = NullValueHandling.Ignore)]
    private sealed class PostgresVerifySpec
    {
        [JsonProperty("query")] public string? Query { get; set; }
        [JsonProperty("expected_value")] public int ExpectedValue { get; set; }
    }
}
